backend/models/user.go

```diff
@@ -1,14 +1,55 @@
 package models
 
+import (
+	"errors"
+	"github.com/go-webauthn/webauthn/webauthn"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
 type User struct {
 	Common
-	Name    string `json:"name"`
-	Email   string `json:"email"`
-	IsAdmin bool   `json:"is_admin,default;False"`
+	Name       string    `json:"name"`
+	Email      string    `gorm:"unique" json:"email"`
+	IsAdmin    bool      `json:"is_admin,default;False"`
+	WebauthnId uuid.UUID `gorm:"type:uuid" json:"webauthnId"`
 }
 
 type UserReq struct {
 	Name    string `json:"name,omitempty"`
 	Email   string `json:"email,omitempty"`
-	IsAdmin bool   `json:"is_admin,string,default;False"`
+	IsAdmin bool   `json:"is_admin,default;False"`
+}
+
+func (u *User) BeforeCreate(tx *gorm.DB) (err error) {
+	webauthnId, err := uuid.NewUUID()
```

Comment:
I just generate a uuid for each user to use for webauthn because it requires a byte array,